src/desktop/components/promotion-card/promotion-card.tsx

```diff
@@ -36,12 +35,20 @@ export interface PromotionCardProps extends HTMLAttributes<HTMLDivElement> {
   'data-testid'?: string;
 }
 
-const PromotionName: Record<PromotionType, string> = {
+const PROMOTION_NAME: Readonly<Record<PromotionType, string>> = {
   gift: 'Подарок',
   special: 'Спецпредложение',
   '3-by-2': '3 по цене 2',
   'volume-discount': 'Скидка за объём',
-} as const;
+};
+
+/**
+ * Проверяет тип акции.
+ * @param type Тип.
+ * @return Признак.
+ */
+const isKnownPromotionType = (type?: string): type is PromotionType =>
+  Boolean(type && Object.keys(PROMOTION_NAME).includes(type));
```

Comment:
Можно просто
```suggestion
  Boolean(type && PROMOTION_NAME[type]);
```
Без преобразования объекта в массив и поиска по массиву

Reply:
а если добавить ```type in PROMOTION_NAME &&```?

Reply:
тоже ожидают строку, кроме как `type guard` особых вариантов то и нет